client/src/router/authUtils.js

```diff
@@ -1,16 +1,49 @@
 import axios from "axios";
 import config from "../../config/config.json";
-import Cookies from "js-cookie";
+import { jwtDecode } from "jwt-decode";
 
-export const checkCookie = async () => {
-  const token = Cookies.get("token");
+const isCookieValid = (cookie, signInTime) => {
+  const token = jwtDecode(cookie);
 
-  if (token != null || token != undefined) {
-    const res = await axios.get(`${config.backend_url}/api/verifyUser`, {
-      withCredentials: true,
-    });
+  if (token.user_type && token.user_type == "regular") {
+    //check hashed_email field exists in token, if
+    //user_id field is Number type and if iat is equal
+    //to sign in time set by auth page
+    if (
+      token.hashed_email &&
+      typeof token.user_id === Number &&
+      token.iat == signInTime
+    )
+      return true;
+  } else if (token.user_type && token.user_type == "guest") {
+    //check if user_id field starts with sess
+    //and if iat is equal to sign in time
+    if (
+      token.user_id &&
+      token.user_id.startsWith("sess") &&
+      token.iat == signInTime
+    )
+      return true;
+  }
+  return false;
+};
 
-    if (res.status === 200) return true;
+export const checkCookie = async (cookie, signInTime) => {
+  if (isCookieValid(cookie, signInTime)) {
+    //if cookie is valid but 1 hour since logged in then verify token,
+    //else let them pass
+    const currTime = Date.now();
+    const timePassed = currTime - signInTime;
+    if (timePassed > 3_600_000) {
```

Comment:
call to `verifyUser` won't refresh user jwt, which means jwt issue time won't change and each refresh after 1 hr passed will make a call to `verifyUser`.

I think we should stop comparing jwt issue time with `signInTime`, so we can update `signInTime` to current time after each call to `verifyUser`.

we can still use jwt issue time for the first `signInTime`, that won't make a different